sdk/resources/Azure.ResourceManager.Resources/src/Generated/Extensions/TenantExtensions.cs

```diff
@@ -73,7 +73,7 @@ public static Response<Deployment> GetDeployment(this Tenant tenant, string depl
         /// <param name="template"> The template provided to calculate hash. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="template"/> is null. </exception>
-        public static async Task<Response<TemplateHashResult>> CalculateTemplateHashDeploymentAsync(this Tenant tenant, object template, CancellationToken cancellationToken = default)
+        public static async Task<Response<TemplateHashResult>> CalculateTemplateHashDeploymentAsync(this Tenant tenant, BinaryData template, CancellationToken cancellationToken = default)
```

Comment:
The name of this method reads weird, maybe we need to rename this operation?

Reply:
This name comes from operationId `Deployments_CalculateTemplateHash` and transformed to `CalculateTemplateHashDeployment` by the generator as extension method name. `CalculateDeploymentTemplateHash` is better.

Reply:
Resolving this here: https://github.com/Azure/azure-sdk-for-net/pull/27605